x-pack/metricbeat/module/aws/awshealth/awshealth.go

```diff
@@ -0,0 +1,437 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package awshealth
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"time"
+
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/health"
+	"github.com/aws/aws-sdk-go-v2/service/health/types"
+
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/x-pack/metricbeat/module/aws"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+const metricsetName = "awshealth"
+
+var (
+	locale     = "en"
+	maxResults = int32(10)
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host is defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet(aws.ModuleName, metricsetName, New,
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	*aws.MetricSet
+	logger *logp.Logger
+	Config Config `config:"aws_health_config"`
+}
+
+// Config holds the configuration specific for aws-awshealth metricset
+type Config struct {
+	EventARNPattern []string `config:"event_arns_pattern"`
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+
+	logger := logp.NewLogger(metricsetName)
+	metricSet, err := aws.NewMetricSet(base)
+	if err != nil {
+		return nil, fmt.Errorf("error creating aws metricset: %w", err)
+	}
+
+	cfgwarn.Beta("The aws:awshealth metricset is beta.")
+
+	config := struct {
+		Config Config `config:"aws_health_config"`
+	}{}
+
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	return &MetricSet{
+		MetricSet: metricSet,
```

Comment:
IMO, we are overusing the name `metricset`, in that we are naming the struct `MetricSet`, embedding the field `*aws.MetricSet` which would again be accessed using `.MetricSet`, and the value that gets assigned, the same. And then we have an interface `mb.MetricSet`. Although they have different meanings under different context, just looking at this particular statement, it doesn't look as readable as something like(and it's just an example):
``` go
type MetricsStateInfo struct {
    *aws.MetricSet
    logger *logp.Logger
    Config Config `config:"aws_health_config"`
}
```

imagine having to reference fields like `metricSet.MetricSet`. This not only is more verbose, but is most likely to be caught by linters as a bad practice. Please consider renaming if possible.

Reply:
++